python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -174,8 +174,7 @@ def _attribute_filter__set__(self, name, value):
         if value != "" and value is not None:
             #   Do some checking of the filenames in a subprocess
             if name == 'lfn':
-                if self.namePattern == '':
-                    self.namePattern = os.path.basename(value)
+                self.namePattern = os.path.basename(value)
```

Comment:
To be honest, this is probably more my lack of understanding of how `namePattern` should be used as I've always found it confusing and possibly contradictory in the code. If things are work then I'm happy 😄